core/src/main/java/fr/sncf/osrd/utils/DoubleRangeMap.java

```diff
@@ -1,13 +1,13 @@
 package fr.sncf.osrd.utils;
 
-import java.util.HashMap;
-import java.util.TreeMap;
+import java.util.*;
 
 public class DoubleRangeMap extends TreeMap<Double, Double> {
```

Comment:
I didn't check in details but Guava has a `RangeMap`. Couldn't we use it instead?

Reply:
Oh this sounds nice

Reply:
Actually this is a fairly large change, and as the API isn't exactly the same we need to be careful not to break things. I'm especially afraid about `addCurvesToGradients`.

I think we should do this, but maybe later on in a different PR. And once this method is better tested.